shiny/ui/_include_helpers.py

```diff
@@ -265,4 +338,111 @@ def read_utf8(path: Path | str) -> str:
         return f.read()
 
 
+def check_bootstrap_compatibility(
+    theme: Path | str | HTMLDependency,
+    bs_version: Optional[Version | str] = None,
+) -> None:
+    if isinstance(theme, HTMLDependency):
+        check_bootstrap_compatibility_dep(theme, bs_version)
+        return
+
+    if bs_version is not None:
+        check_bootstrap_compatibility_version(bs_version)
+        return
+
+    if isinstance(theme, (Path, str)):
+        check_bootstrap_compatibility_file(theme)
+        return
+
+    raise RuntimeError(
+        "Invalid theme argument. Must be a Path, str, or HTMLDependency."
+    )
+
+
+def check_bootstrap_compatibility_dep(
+    theme: HTMLDependency,
+    bs_version: Optional[Version | str] = None,
+) -> None:
+    if bs_version is not None:
+        check_bootstrap_compatibility_version(
+            bs_version=bs_version,
+            name=theme.name,
+            version=theme.version,
+        )
+
+    if hasattr(theme, "bs_version"):
+        bs_version = getattr(theme, "bs_version", None)
+        if bs_version is not None and isinstance(bs_version, (str, Version)):
+            check_bootstrap_compatibility_version(
+                bs_version=bs_version,
+                name=theme.name,
+                version=theme.version,
+            )
+
+    source_dir = theme.source_path_map()["source"]
+
+    if not theme.stylesheet:
+        return
+
+    sheet = theme.stylesheet[0]["href"]
+
+    check_bootstrap_compatibility_file(Path(source_dir) / Path(sheet).name)
+
+
+def check_bootstrap_compatibility_file(theme: str | Path):
+    theme = Path(theme)
+
+    if theme.suffix != ".css":
+        warn(
+            "Cannot check Bootstrap compatibility of non-CSS files. "
+            + f"Skipping: {theme}",
+        )
+
+    version = ""
+    with open(theme, "r", encoding="utf-8") as f:
+        for line in f:
+            if re.search(r"\* Bootstrap\s+v", line):
```

Comment:
This search might not work for shinyswatch https://github.com/posit-dev/py-shinyswatch/blob/main/shinyswatch/bsw5/cerulean/bootswatch.min.css

Reply:
I explicitly tested with Shinyswatch, the goal is to find this line:

https://github.com/posit-dev/py-shinyswatch/blob/1f13e89bf00804e9906cee9e7137353212542487/shinyswatch/bsw5/cerulean/bootswatch.min.css#L5C1-L5C23

Reply:
Sorry, I thought the `\*` was literal when I first read it 🤦